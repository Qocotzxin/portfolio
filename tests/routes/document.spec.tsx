import Document from "@pages/_document";
import { render, screen } from "@testing-library/react";
import { PropsWithChildren } from "react";

jest.mock("next/document", () => ({
  Html: ({ children }: PropsWithChildren) => (
    <div>
      <span>Html</span>
      {children}
    </div>
  ),
  Head: () => <div>Head</div>,
  Main: () => <div>Main</div>,
  NextScript: () => <div>NextScript</div>,
}));

jest.mock("@chakra-ui/react", () => {
  const actual = jest.requireActual("@chakra-ui/react");

  return {
    ...actual,
    ColorModeScript: ({ children }: PropsWithChildren) => (
      <div>
        <span>ColorModeScript</span>
        {children}
      </div>
    ),
  };
});

describe("_document", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<Document />);
    expect(container).toMatchSnapshot();
  });

  it("Should render next/document sections, the body and color mode script.", () => {
    render(<Document />);
    expect(screen.getByText(/Html/)).toBeInTheDocument();
    expect(screen.getByText(/Head/)).toBeInTheDocument();
    expect(screen.getByText(/Main/)).toBeInTheDocument();
    expect(screen.getByText(/NextScript/)).toBeInTheDocument();
    expect(screen.getByText(/ColorModeScript/)).toBeInTheDocument();
  });
});
