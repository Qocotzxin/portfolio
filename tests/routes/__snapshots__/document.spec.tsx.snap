// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_document Should match snapshot. 1`] = `
<div>
  <div>
    <span>
      Html
    </span>
    <div>
      Head
    </div>
    <body>
      <div>
        <span>
          ColorModeScript
        </span>
      </div>
      <div>
        Main
      </div>
      <div>
        NextScript
      </div>
    </body>
  </div>
</div>
`;
